src/components/contexts/connection-handler.tsx

```diff
@@ -0,0 +1,60 @@
+import { Typography } from "@mui/material";
+import GenericSnackbar from "components/generic/generic-snackbar";
+import strings from "localization/strings";
+import React, { useEffect, useState } from "react";
+
+/**
+ * ConnectionHandler component
+ */
+const ConnectionHandler: React.FC = ({ children }) => {
+  const [ connectionLost, setConnectionLost ] = useState<boolean>(false);
+
+  /**
+   * Renders connection lost alert
+   */
+  const renderConnectionLostAlert = () => (
+    <GenericSnackbar
+      open={ connectionLost }
+      onClose={ () => setConnectionLost(!connectionLost) }
+      severity="error"
+    >
+      <>
+        <Typography variant="h5">{ strings.errorHandling.formScreen.noConnection }</Typography>
+        <Typography variant="caption">{ strings.errorHandling.formScreen.noConnectionHelper }</Typography>
+      </>
+    </GenericSnackbar>
+  );
```

Comment:
Move this to bottom, before component return.

Reply:
Done